src/extension/ncdfgenerator/src/main/java/au/org/emii/ncdfgenerator/FeatureTypeFactory.java

```diff
@@ -0,0 +1,110 @@
+package au.org.emii.ncdfgenerator;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.sql.Connection;
+import java.sql.SQLException;
+
+import org.geotools.jdbc.VirtualTable;
+import org.geotools.data.Query;
+import org.geotools.data.Transaction;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.geotools.jdbc.JDBCDataStore;
+import org.opengis.feature.type.*;
+import org.geotools.data.simple.SimpleFeatureIterator;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+
+import com.vividsolutions.jts.geom.Geometry;
+
+public class FeatureTypeFactory {
+    // Lifted from: https://github.com/geoserver/geoserver/blob/master/src/web/core/src/main/java/org/geoserver/web/data/layer/SQLViewAbstractPage.java#L404
```

Comment:
What I was thinking was separating out the code you have added from what you have lifted to make it clearer what was lifted (see example below).  Not sure you would just make this a public method in GeoServer - the common bits would probably need to be factored out.   Its not a biggie - just a thought I had when trying to reconcile your changes with the geoserver code - I had to work out what you had lifted and what you had added.

    public static SimpleFeatureType guessFeatureType(JDBCDataStore store, String virtualTableDef, boolean guessGeometrySrid) throws IOException {

        String vtName = "vt";

        try {
            return guessFeatureType(store, vtName, guessGeometrySrid);
       }
       finally {
           store.dropVirtualTable(vtName);
       }
    }

   
   // Lifted from: https://github.com/geoserver/geoserver/blob/master/src/web/core/src/main/java/org/geoserver/web/data/layer/SQLViewAbstractPage.java#L404
    public static SimpleFeatureType guessFeatureType(JDBCDataStore store, String vtName, boolean guessGeometrySrid) throws IOException {
        SimpleFeatureType base = store.getSchema(vtName);
        List<String> geometries = new ArrayList<String>();
        ...